assets/javascripts/discourse/routes/discourse-ai-shared-conversation-show.js

```diff
@@ -1,8 +1,19 @@
+import { service } from "@ember/service";
 import DiscourseRoute from "discourse/routes/discourse";
 
-export default DiscourseRoute.extend({
+export default class DiscourseAiSharedConversationShowRoute extends DiscourseRoute {
+  @service currentUser;
+
   beforeModel(transition) {
-    window.location = transition.intent.url;
+    if (this.currentUser?.user_option?.external_links_in_new_tab) {
+      window.open(transition.intent.url, "_blank");
+    } else {
+      this.redirect(transition.intent.url);
+    }
     transition.abort();
-  },
-});
+  }
+
+  redirect(url) {
+    window.location = url;
```

Comment:
Minor nit, but why bother with this function when it's only used on one line and it's also a one liner?

Reply:
Otherwise we can't test it. It's really hard to mock/stub `window.location` 🤷‍♂️